demo/vue3js.html

```diff
@@ -1,88 +1,97 @@
-
-          // Use an arrow function so that `this` is bound to the Vue instance. Alternatively, use a custom Vue directive on the `.grid-stack` container element: https://vuejs.org/v2/guide/custom-directive.html
-          this.grid.on("dragstop", (event, element) => {
-            const node = element.gridstackNode;
-            // `this` will only access your Vue instance if you used an arrow function, otherwise `this` binds to window scope. see https://hacks.mozilla.org/2015/06/es6-in-depth-arrow-functions/
-            this.info = `you just dragged node #${node.id} to ${node.x},${node.y} – good job!`;
-          });
-        },
-        methods: {
-          addNewWidget: function () {
-            const node = this.$options.items[this.count] || {
-              x: Math.round(12 * Math.random()),
-              y: Math.round(5 * Math.random()),
-              w: Math.round(1 + 3 * Math.random()),
-              h: Math.round(1 + 3 * Math.random()),
-            };
-            node.id = node.content = String(this.count++);
-            this.grid.addWidget(node);
-          },
-        },
-      }).mount("#app");
-    </script>
-  </body>
-</html>
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Gridstack.js Vue integration example</title>
+    <link rel="stylesheet" href="demo.css" />
+    <script src="../dist/gridstack-h5.js"></script>
+  </head>
+  <body>
+    <main id="app">
+      <h1>How to integrate GridStack.js with Vue.js</h1>
+      <p>
+        As with any virtualDOM-based framework, you need to check if Vue has
+        rendered the DOM (or any updates to it) <strong>before</strong> you
+        initialize GridStack or call its methods. As a basic example, check this
+        component's <code>mounted</code> hook.
+      </p>
+      <p>
+        If your app requires more complex render logic than the inline template
+        in `addWidget`, consider
+        <a
+          href="https://github.com/gridstack/gridstack.js/tree/develop/doc#makewidgetel"
+          >makeWidget</a
+        >
+        to let Vue deal with DOM rendering.
+      </p>
+      <button type="button" @click="addNewWidget()">Add Widget</button> {{ info
+      }}
+      <section class="grid-stack"></section>
+    </main>
+    <script type="module">
+      import { createApp } from "https://cdn.jsdelivr.net/npm/vue@3.0.0/dist/vue.esm-browser.js";
+      import { ref, onMounted } from "vue";
+
+      createApp({
+        setup() {
+          let count = ref(0);
+          let info = ref("");
+          let grid = null;
+          const items = [
+            { x: 2, y: 1, h: 2 },
+            { x: 2, y: 4, w: 3 },
+            { x: 4, y: 2 },
+            { x: 3, y: 1, h: 2 },
+            { x: 0, y: 6, w: 2, h: 2 },
+          ];
+
+          onMounted(() => {
+            grid = GridStack.init({

```

Comment:
no this.grid needed ? how is that scoped then ????
(don't want it global but this Vue component...)

I would insert the comment back
// Provides access to the GridStack instance across the Vue component.

Reply:
From my understanding you got access to the grid within the setup function at all time. So it is only accessible for the Vue component itself. 
So you are able to define functions or vars etc. inside the setup function which are able to work with the grid.

Reply:
ok, did you want to make line ending change and merge into 1 review, then I can accept it. thank you!